comps/texttosql/langchain/src/texttosql.py

```diff
@@ -0,0 +1,328 @@
+﻿# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import os
+from typing import Any, Dict, List, Optional, Sequence, Type, Union
+
+from langchain.agents import create_react_agent
+from langchain.agents.agent import AgentExecutor, RunnableAgent
+from langchain.agents.agent_types import AgentType
+from langchain_community.agent_toolkits.sql.prompt import SQL_PREFIX, SQL_SUFFIX
+from langchain_community.agent_toolkits.sql.toolkit import SQLDatabaseToolkit
+from langchain_community.tools.sql_database.prompt import QUERY_CHECKER
+from langchain_community.tools.sql_database.tool import InfoSQLDatabaseTool, ListSQLDatabaseTool
+from langchain_community.utilities.sql_database import SQLDatabase
+from langchain_core.callbacks import AsyncCallbackManagerForToolRun, BaseCallbackManager, CallbackManagerForToolRun
+from langchain_core.language_models import BaseLanguageModel
+from langchain_core.prompts import BasePromptTemplate, PromptTemplate
+from langchain_core.pydantic_v1 import BaseModel, Field, root_validator
+from langchain_core.tools import BaseTool
+from langchain_huggingface import HuggingFaceEndpoint
+from sqlalchemy.engine import Result
+
+from comps import CustomLogger
+
+generation_params = {
+    "max_new_tokens": 1024,
+    "top_k": 10,
+    "top_p": 0.95,
+    "temperature": 0.01,
+    "repetition_penalty": 1.03,
+    "streaming": True,
+}
+
+
+TGI_LLM_ENDPOINT = os.environ.get("TGI_LLM_ENDPOINT")
+
+llm = HuggingFaceEndpoint(
+    endpoint_url=TGI_LLM_ENDPOINT,
+    task="text-generation",
+    **generation_params,
+)
+
+sql_params = {
+    "max_string_length": 3600,
+}
+
+logger = CustomLogger("comps-texttosql")
+logflag = os.getenv("LOGFLAG", False)
+
+# https://github.com/langchain-ai/langchain/issues/23585
```

Comment:
this github issue here is known issue to aware of?

Reply:
Yes, it's still an open issue in langchain due to which the tools do not function correctly. 
I have added the reference to the bug and incorporated the solution (by creating custom tools) to overcome the issue.